### PR TITLE
Review: move sorald to subfolder

### PR DESCRIPTION
**Pull request by MartinWitt**
As always, any perfect plan fails on some small things you forgot :D. Maven more or less needs a top-level parent for multimodule projects. I thought maven has the same features as Gradle for multimodule projects. This PR changes the sorald-parent to a top-level pom and moves all files from `/src` to a Sorald subproject. I split the PR in two commits on is only file movement and the other one is pom changes. Luckily, GitHub shows when a file is moved without changes. I believe the best way to review this is checkout the PR and have a look inside your IDE.

### CODE REVIEW

**Comment by algomaster99** [comment]
I shall take a look at this on Monday :)

File: pom.xml [hunk 1]
```diff
@@ -206,31 +165,6 @@
...
-                            <goal>report</goal>
-                        </goals>
-                        <configuration>
-                            <excludes>
-                                <exclude>sorald/annotations/**/*.class</exclude>
-                            </excludes>
-                        </configuration>
-                    </execution>
-                </executions>
-            </plugin>
```
Comment (algomaster99) [inline pom.xml:233]: Shouldn't dependenices like `picocli` and `jacoco` be in the top-level pom? I think they would be shared by all submodules?

**Comment by algomaster99** [comment]
@MartinWitt I merged `master` with `spi`. You will need to resolve some conflicts here.

File: pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (MartinWitt) [reply pom.xml:233]: No only sorald core (the sorald subfolder) needs the CLI feature currently. JaCoCo could be added to top level, but then we would force all projects implementing the API to use JaCoCo. I'm unsure if this is a great result.

**Pull request merged**